vllm/engine/arg_utils.py

```diff
@@ -1225,19 +1225,24 @@ def _override_v1_engine_args(self, usage_context: UsageContext) -> None:
         """
         assert envs.VLLM_USE_V1, "V1 is not enabled"
 
+        # V1 always uses chunked prefills.
+        self.enable_chunked_prefill = True
+        # When no user override, set the default values based on the usage
+        # context.
+        # TODO(woosuk): Tune the default values for different hardware.
         if self.max_num_batched_tokens is None:
-            # When no user override, set the default values based on the
-            # usage context.
             if usage_context == UsageContext.LLM_CLASS:
-                logger.warning("Setting max_num_batched_tokens to 8192 "
-                               "for LLM_CLASS usage context.")
-                self.max_num_seqs = 1024
                 self.max_num_batched_tokens = 8192
             elif usage_context == UsageContext.OPENAI_API_SERVER:
-                logger.warning("Setting max_num_batched_tokens to 2048 "
-                               "for OPENAI_API_SERVER usage context.")
-                self.max_num_seqs = 1024
                 self.max_num_batched_tokens = 2048
+            logger.warning(
+                "Setting max_num_batched_tokens to %d for %s usage context.",
+                self.max_num_batched_tokens, usage_context.value)
+        # NOTE(woosuk): Increase max_num_seqs since the default value (256) is
+        # too small to achieve the best performance in V1.
+        self.max_num_seqs = 1024
```

Comment:
I'm afraid this is not optimal if you're serving large models on smaller GPUs.

Reply:
Right. Let me find a way to allow overriding it.

Reply:
agree with @comaniac . this value looks like overfitting to the benchmarking setting you run (H100 with 8B model). IMO we should directly use the value without overriding it, and let users specify it.

Reply:
@comaniac fixed in https://github.com/vllm-project/vllm/pull/11061/commits/2dbdb58e1829eb4d5a48044c14aced88b504e9e6